packages/duckietown/include/duckietown/dtros/dtpublisher.py

```diff
@@ -77,3 +93,29 @@ def publish(self, *args, **kwargs):
             self._tick_frequency()
             # call super publish
             super(DTPublisher, self).publish(*args, **kwargs)
+
+    def register_subscribers_changed_cb(self, cb_fun):
+        """
+        Registers a callback that will be called any time the number of subscribers change.
+        The callbacks should take a single argument which is the publisher object itself.
+
+        Args:
+            cb_fun: the callback function
+        """
+        self._subs_changed_callbacks.append(cb_fun)
```

Comment:
use a set instead of a list, it gives you the benefit of avoiding double subscription of the same callback

Reply:
done.